ast/src/main/java/com/graphicsfuzz/common/tool/PrettyPrinterVisitor.java

```diff
@@ -460,23 +464,22 @@ public void visitWhileStmt(WhileStmt whileStmt) {
 
   @Override
   public void visitForStmt(ForStmt forStmt) {
-    out.append(indent() + "for(" + newLine());
-    out.append("    ");
+    insideForStatement = true;
+    out.append(indent() + "for(");
     visit(forStmt.getInit());
-    out.append("    " + indent());
+    out.append(" ");
     if (forStmt.hasCondition()) {
       visit(forStmt.getCondition());
     }
-    out.append(";" + newLine());
-    out.append("    " + indent());
+    out.append("; ");
     if (forStmt.hasIncrement()) {
       visit(forStmt.getIncrement());
     }
-    out.append(newLine());
-    out.append(indent() + ")" + newLine());
+    out.append(" )" + newLine());
```

Comment:
Please remove the space in " )", so that we don't have a space between the end of the for loop header and its closing ")".

Reply:
Done.